engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java

```diff
@@ -370,6 +371,13 @@ public void resumeProcessing(final int partitionId) {
                 }));
   }
 
+  public void awaitProcessingOf(final Record<?> record) {
+    final var recordPosition = record.getPosition();
+
+    Awaitility.await("await the record to be processed")
```

Comment:
Would be cool if when it fails we can see get more context (e.g. actual record position, expected record position). Maybe we should use an untilAsserted then? Or something with the alias?

```suggestion
    Awaitility.await("await the record at position " + recordPosition + " has been processed")
```

Though this doesn't give us the actual...maybe the untilAsserted gives us a better error message?

```suggestion
    Awaitility.await("await the record at position " + recordPosition + " has been processed")
      .untilAsserted(() -> assertThat(getLastProcessedPosition()).as("the last processed position is >= the expected record position").isGreaterThanOrEqualTo(recordPosition))
```

I didn't test that, but maybe when it fails it gives us more info